newrelic/hooks/mlmodel_openai.py

```diff
@@ -24,14 +24,53 @@
 from newrelic.core.config import global_settings
 
 
+def openai_error_attributes(exception, request_args):
+    # 'id' attribute will always be 'None' in this case.
+    # 'completion_id' is generated at the completion of
+    # the request and all of these error types occur
+    # before the response is generated.
+    #
+    api_key_LFD = None
+    if openai.api_key:
+        api_key_LFD = f"sk-{openai.api_key[-4:]}"
+
+    error_code = "None"
+    status_code = getattr(exception, "http_status", "None")
+    if status_code and status_code >= 400 and status_code < 600:
+        error_code = status_code
+
+    error_attributes = {
+        "id": "None",
+        "api_key_last_four_digits": api_key_LFD,
+        "request.model": request_args.get("model") or request_args.get("engine", "None"),
+        "temperature": request_args.get("temperature", "None"),
+        "max_tokens": request_args.get("max_tokens", "None"),
+        "vendor": "openAI",
+        "ingest_source": "Python",
+        "organization": getattr(exception, "organization", "None"),
+        "number_of_messages": len(request_args.get("messages", [])),
+        "status_code": status_code,
+        "error_message": getattr(exception, "_message", "None"),
+        "error_type": exception.__class__.__name__ or "None",
```

Comment:
Our current notice_error instrumentation already pulls the class off of the exception internally so this attribute is a repeat. The difference is that our instrumentation includes the module path in the class name but I think that's fine here.
```suggestion
```